mlflow/store/tracking/utils/sql_trace_metrics_utils.py

```diff
@@ -0,0 +1,326 @@
+from datetime import datetime, timezone
+from typing import TypedDict
+
+from sqlalchemy import Column, and_, func, literal_column
+from sqlalchemy.orm.query import Query
+
+from mlflow.entities.trace_metrics import (
+    AggregationType,
+    MetricAggregation,
+    MetricDataPoint,
+    MetricViewType,
+)
+from mlflow.exceptions import MlflowException
+from mlflow.store.tracking.dbmodels.models import SqlTraceInfo, SqlTraceTag
+from mlflow.tracing.constant import TraceTagKey
+
+
+class TraceMetricsConfig(TypedDict):
+    """
+    Configuration for traces metrics.
+
+    Args:
+        aggregation_types: Supported aggregation types to apply to the metrics.
+        dimensions: Supported dimensions to group metrics by.
+        filter_fields: Supported fields to filter on.
+    """
+
+    aggregation_types: set[AggregationType]
+    dimensions: set[str]
+    # TODO: support this
+    filter_fields: set[str] | None
+
+
+# metric_name -> TraceMetricsConfig mapping for traces
+TRACES_METRICS_CONFIGS: dict[str, TraceMetricsConfig] = {
+    "trace": {
+        "aggregation_types": {AggregationType.COUNT},
+        "dimensions": {"name", "status"},
+        "filter_fields": None,
+    },
+    "latency": {
+        "aggregation_types": {AggregationType.AVG, AggregationType.PERCENTILE},
+        "dimensions": {"name"},
+        "filter_fields": None,
+    },
+}
+
+# TODO: add spans and assessments metrics configs
+VIEW_TYPE_CONFIGS: dict[MetricViewType, dict[str, TraceMetricsConfig]] = {
+    MetricViewType.TRACES: TRACES_METRICS_CONFIGS,
+}
+
+TIME_BUCKET_LABEL = "time_bucket"
+
+
+def get_percentile_aggregation(db_type: str, percentile_value: float, column):
```

Comment:
nit: if we have some expected types for this, should we use `Literal`?

```suggestion
def get_percentile_aggregation(db_type: Literal["postgresql", "mssql", "mysql", "sqlite"], percentile_value: float, column):
```

Reply:
This is internal function so I think it's fine to keep as string, e.g. if we want to drop mssql in the future then we don't need to update this :)